eng/scripts/CodeChecks.ps1

```diff
@@ -94,6 +96,45 @@ try {
         & $PSScriptRoot\Export-API.ps1 -ServiceDirectory $ServiceDirectory -SDKType $SDKType -SpellCheckPublicApiSurface:$SpellCheckPublicApiSurface
     }
 
+    Write-Host "Validating installation instructions"
+    Join-Path "$PSScriptRoot/../../sdk" $ServiceDirectory  `
+        | Resolve-Path `
+        | % { Get-ChildItem $_ -Filter "README.md" -Recurse } `
+        | % {
+            $readmePath = $_
+            $readmeContent = Get-Content $readmePath
+            if ($readmeContent -Match "dotnet add .*--version.*$")
+            {
+                LogError "Specific versions should not be specified in the installation instructions. For beta versions, include the --prelease flag."
+                break
+            }
+            
+            if ($readmeContent -Match "dotnet add")
+            {
+                $changelogPath = Join-Path $(Split-Path -Parent $readmePath) "CHANGELOG.md"
+                $changeLogEntries = Get-ChangeLogEntries -ChangeLogLocation $changelogPath
+                $hasGa = $false;
+                foreach ($key in $changeLogEntries.Keys)
+                {
+                    $entry = $changeLogEntries[$key]
+                    if ($entry.ReleaseStatus -ne "(Unreleased)" -and $entry.ReleaseVersion -notmatch "beta")
```

Comment:
Can't we sniff this from the package feed common function, and would that be more reliable?

Reply:
The problem I see there is that we'd have to do two separate PRs - one to update changelog, and then a separate to update README after the release.

Reply:
Yeah, I see your point.   I was thinking the package feed would more reliably detect if there was already a GA, but it would fail to detect when this is the first ever GA until after the package was published - which is too late for the README riding along with it.